extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js

```diff
@@ -183,6 +192,78 @@ class ReplicationStatusProcessor {
                          { entry: sourceEntry.getLogInfo() });
         return process.nextTick(done);
     }
+
+    /**
+     * Handle ProbeServer liveness check
+     *
+     * @param {http.HTTPServerResponse} res - HTTP Response to respond with
+     * @param {Logger} log - Logger
+     * @returns {string} Error response string or undefined
+     */
+    handleLiveness(res, log) {
+        const verboseLiveness = {};
+        // track and return all errors in one response
+        const responses = [];
+
+        if (this._consumer === undefined || this._consumer === null) {
+            verboseLiveness.consumer = constants.statusUndefined;
+            responses.push({
+                component: 'Consumer',
+                status: constants.statusUndefined,
+            });
+        } else if (!this._consumer._consumerReady) {
+            verboseLiveness.consumer = constants.statusNotReady;
+            responses.push({
+                component: 'Consumer',
+                status: constants.statusNotReady,
+            });
+        } else {
+            verboseLiveness.consumer = constants.statusReady;
+        }
+
+        if (this._FailedCRRProducer === undefined || this._FailedCRRProducer === null ||
+            this._FailedCRRProducer._producer === undefined ||
+            this._FailedCRRProducer._producer === null) {
+            verboseLiveness.failedCRRProducer = constants.statusUndefined;
+            responses.push({
+                component: 'Failed CRR Producer',
+                status: constants.statusUndefined,
+            });
+        } else if (!this._FailedCRRProducer._producer.isReady()) {
+            verboseLiveness.failedCRRProducer = constants.statusNotReady;
+            responses.push({
+                component: 'Failed CRR Producer',
+                status: constants.statusNotReady,
+            });
+        } else {
+            verboseLiveness.failedCRRProducer = constants.statusReady;
+        }
+
+        log.debug('verbose liveness', verboseLiveness);
+
+        if (responses.length > 0) {
+            return JSON.stringify(responses);
```

Comment:
won't `return` here break?

Reply:
Currently the arsenal probe server will handle string responses and finish the response for you ( it has been proposed to change that but that is what it is currently doing )